core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/TransferProcessCommandExtension.java

```diff
@@ -36,12 +35,7 @@
 
     @Override
     public void initialize(ServiceExtensionContext context) {
-        CommandHandlerRegistry registry = context.getService(CommandHandlerRegistry.class, true);
-        if (registry == null) {
-            registry = new CommandHandlerRegistryImpl();
-            context.registerService(CommandHandlerRegistry.class, registry);
-        }
-
+        CommandHandlerRegistry registry = context.getService(CommandHandlerRegistry.class);
```

Comment:
Should be injected via `@Inject`

Reply:
OK, let's do that in a subsequent PR because there is some other cleanup I want to do here that is related.